pkg/coscheduling/core/core.go

```diff
@@ -112,6 +116,33 @@ func (pgMgr *PodGroupManager) PreFilter(ctx context.Context, pod *corev1.Pod) er
 		return fmt.Errorf("cannot found engough pods, "+
 			"current pods number: %v, minMember of group: %v", len(pods), pg.Spec.MinMember)
 	}
+
+	if pg.Spec.MinResources == nil {
+		return nil
+	}
+
+	// TODO(cwdsuzhou): This resource check may not always pre-catch unschedulable pod group.
+	// It only tries to PreFilter resource constraints so even if a PodGroup passed here,
+	// it may not necessarily pass Filter due to other constraints such as affinity/taints.
+	if _, ok := pgMgr.permittedPG.Get(pgFullName); ok {
+		return nil
+	}
+
+	nodes, err := pgMgr.snapshotSharedLister.NodeInfos().List()
+	if err != nil {
+		return err
+	}
+
+	minResources := *pg.Spec.MinResources
+	podQuantity := resource.NewQuantity(int64(pg.Spec.MinMember), resource.DecimalSI)
+	minResources[corev1.ResourcePods] = *podQuantity
+	err = pgMgr.CheckClusterResource(nodes, minResources)
+	if err != nil {
+		klog.Error(err)
```

Comment:
Please prepend the PodGroup name to the error.

Reply:
Done